python/ray/util/sgd/v2/callbacks/logging.py

```diff
@@ -87,3 +121,125 @@ def handle_result(self, results: Optional[List[Dict]]):
+                 workers_to_log: int = 0) -> None:
+        return super().__init__(
+            logdir=logdir, filename=filename, workers_to_log=workers_to_log)
+
+    def _validate_workers_to_log(self, workers_to_log) -> int:
+        if isinstance(workers_to_log, Iterable):
+            workers_to_log = list(workers_to_log)
+            if len(workers_to_log) > 1:
+                raise ValueError(
+                    f"{self.__class__.__name__} only supports logging "
+                    "from a single worker.")
+            elif len(workers_to_log) < 1:
+                raise ValueError(
+                    "At least one worker must be specified in workers_to_log.")
+            workers_to_log = workers_to_log[0]
+        if not isinstance(workers_to_log, int):
+            raise TypeError("workers_to_log must be an integer.")
+        return workers_to_log
+
+
+class TBXLoggerCallback(SGDSingleFileSingleWorkerLoggingCallback):
+    """Logs SGD results in TensorboardX format.
+
+    Args:
+        logdir (Optional[str]): Path to directory where the results file
+            should be. If None, will be set by the Trainer.
+        filename (Optional[str]): Ignored.
+        workers_to_log (int): Worker index to log. By default, will log the
+            worker with index 0.
+    """
+
+    _default_filename: Union[str, Path] = None
+    VALID_SUMMARY_TYPES: Tuple[type] = (int, float, np.float32, np.float64,
+                                        np.int32, np.int64)
+    IGNORE_KEYS: Set[str] = {PID, TIMESTAMP, TIME_TOTAL_S, TRAINING_ITERATION}
+
+    def __init__(self,
+                 logdir: Optional[str] = None,
+                 filename: Optional[str] = None,
+                 workers_to_log: int = 0) -> None:
+        if filename:
+            warnings.warn("filename argument is ignored by TBXLoggerCallback."
+                          "Set filename to None to avoid this warning.")
+
+        super().__init__(
+            logdir=logdir, filename=filename, workers_to_log=workers_to_log)
+
+    def _create_log_path(self, logdir_path: Path, filename: Path) -> Path:
+        return logdir_path
+
+    def start_training(self, logdir: str, **info):
+        super().start_training(logdir, **info)
+
+        try:
+            from tensorboardX import SummaryWriter
+        except ImportError:
+            if log_once("tbx-install"):
+                warnings.warn(
+                    "pip install 'tensorboardX' to see TensorBoard files.")
+            raise
+
+        self._file_writer = SummaryWriter(self.log_path, flush_secs=30)
+
+    def handle_result(self, results: Optional[List[Dict]], **info):
+        result = results[self._workers_to_log]
+        step = result[TRAINING_ITERATION]
+        result = {k: v for k, v in result.items() if k not in self.IGNORE_KEYS}
+        flat_result = flatten_dict(result, delimiter="/")
+        path = ["ray", "sgd"]
+
+        # same logic as in ray.tune.logger.TBXLogger
```

Comment:
Probably out of scope for this PR, but is this logic something that can be abstracted away into `ml_utils` and then just called by both the Tune logger and SGD logger? Same for the Json logger

Reply:
Yeah, I was thinking about that. It could be made generic. I'll make an issue and later, a follow-up PR.